### PR TITLE
Add monitor memory search

### PR DESCRIPTION
**Pull request by cedricp**
Hi,
I added a memory search functionnality. I did it to hack some games and finally finish them !
Here's a quick doc : 
- launch the game and check for lives number
- launch monitor (F2) execute **ms <lives#>**
- return to game
- die
- relaunch monitor and execute **mr <lives#-1>**
- show memory addresses with **p** command

At this point you should get one or more memory match (refine more if necessary)
now you should know where the lives counter is in memory, just do a **mm <memory_address> <new lives#>** to set new lives number and you're done.

new commands : 
- ms [memory search]
- mr [memory search refine]
- p [print found memory locations]

The search is made in free ram (0x500-0x97FF)

Hope you'll like it :)

### CODE REVIEW

File: ReadMe.txt [hunk 1]
```diff
@@ -301,7 +301,7 @@ Commands:
   mw <addr>             - Memory watch at addr
   ms <value>            - Search value in RAM memory
   mr <value>            - Refine search with new value
-  mp                    - Print memory search addresses
+  p                     - Print memory search addresses
```
Comment (iss000) [inline ReadMe.txt:304]: Here I vote for 'mp' instead of single 'p'. This way it will be easier to remember.

File: monitor.c [hunk 2]
```diff
@@ -2940,24 +2941,60 @@ SDL_bool mon_cmd( char *cmd, struct machine *oric, SDL_bool *needrender )
...
           if( !mon_getnum( oric, &w, cmd, &i, SDL_TRUE, SDL_FALSE, SDL_FALSE, SDL_TRUE ) )
           {
             mon_str( "Value expected" );
             break;
           }
+          memory_search_pos = 0;
           uint16_t* collection = memory_search;
           memory_hits = 0;
-          for (int j=0x500; j < 0x97FF; j++){
+          for (int j=0x500; j < 0x9800; j++){
```
Comment (iss000) [inline monitor.c:2997]: This is good one ;).

File: ReadMe.txt (hunk 1 above)
Comment (cedricp) [reply ReadMe.txt:304]: I agree, but I couldn't make it work with command repetition :(

**Review by iss000 — CHANGES_REQUESTED** [review]
Nice addition! Pls, review my suggestions, so we can merge changes.

**Review by iss000 — COMMENTED** [review]
Cool!

**Comment by iss000** [comment]
> @ALL: Let's test the new addition :)

Well, it works for me! IMHO it's useful and nice.
Maybe possible improvement: no need to repeat typing 'mp' to see next page with results but just press enter (i.e. like commands 'd' or 'm').

**Comment by cedricp** [comment]
This was working before you asked me to go back to 'mp' command. 
Will try to find a hack for that

**Comment by iss000** [comment]
Now I get the problem :).
OK, possible solution: introduce 'fake' `lastcmd = '+';` and handle it as repeat of what's in the `ibuf`... or something like this...

**Comment by cedricp** [comment]
FYI : Just finished _Xenon1_, ans funny thing, the live counter is at the same memory place as _Zorgon's revenge_ !
Attaching a memory watch to it and removing the decrease operation gave me infinite lives, love it :)
Glad you like it

**Pull request merged**